src/fastcs/datatypes.py

```diff
@@ -60,3 +73,15 @@ class String(DataType[str]):
     @property
     def dtype(self) -> type[str]:
         return str
+
+
+def validate_value(datatype: DataType[T], value: T) -> T:
+    """Validate a value against a datatype."""
+
+    if isinstance(datatype, (Int | Float)):
+        assert isinstance(value, (int | float)), f"Value {value} is not a number"
+        if datatype.min is not None and value < datatype.min:
+            raise ValueError(f"Value {value} is less than minimum {datatype.min}")
+        if datatype.max is not None and value > datatype.max:
+            raise ValueError(f"Value {value} is greater than maximum {datatype.max}")
+    return value
```

Comment:
I think this should be a method of `Datatype` that with default implementation `return value`.